src/main/java/com/epam/selectioncommittee/repository/impl/FacultyRepositoryImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.epam.selectioncommittee.repository.impl;
+
+import com.epam.selectioncommittee.model.Faculty;
+import com.epam.selectioncommittee.repository.FacultyRepository;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@Repository
+@AllArgsConstructor
+public class FacultyRepositoryImpl implements FacultyRepository {
+    private final List<Faculty> list = new ArrayList<>();
+
+    @Override
+    public Faculty createFaculty(Faculty faculty) {
+        list.add(faculty);
+        log.info("Created Faculty ", faculty);
+        return faculty;
+    }
+
+    @Override
+    public List<Faculty> getAllFaculties() {
+        return new ArrayList<>(list);
+    }
+
+    @Override
+    public Faculty getFacultyByName(String name) {
+        log.info("Get faculty with {}", name);
+        return list.stream()
+                .filter(faculty -> faculty.getName().equals(name))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Faculty is not found!"));

```

Comment:
I would recommend to create specific custom exception, like FacultyNotFoundException or EntityNotFoundException instead of RuntimeException

Reply:
Thank you for the recommendations. I have created a custom exception in the next homework (HW_4).